src/network/MissionControlProtocol.cpp

```diff
@@ -190,6 +190,9 @@ void MissionControlProtocol::handleCameraStreamOpenRequest(const json& j) {
 	if (supported_cams.find(cam) != supported_cams.end()) {
 		std::unique_lock<std::shared_mutex> stream_lock(this->_stream_mutex);
 		this->_open_streams[cam] = 0;
+		if (this->_camera_encoders.find(cam) == this->_camera_encoders.end()) {
+			this->_camera_encoders[cam] = std::make_shared<video::H264Encoder>(j["fps"]);
+		}
```

Comment:
This misses the case where the camera stream has been opened and closed in the past, and is being reopened with a different fps.

Since we're using pointers here my earlier advice doesn't apply, so instead of the `if` you can just replace this block with:
```c++
this->_camera_encoders[cam] = std::make_shared<video::H264Encoder>(j["fps"]);
```
This will handle the edge case described above too.